src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -777,6 +782,10 @@ _nns_edge_message_handler (void *thread_data)
     _nns_edge_remove_connection (eh, client_id);
     nns_edge_event_invoke_callback (eh->event_cb, eh->user_data,
         NNS_EDGE_EVENT_CONNECTION_CLOSED, NULL, 0, NULL);
+    if (NNS_EDGE_CONNECT_TYPE_HYBRID == eh->connect_type) {
+      nns_edge_logd ("Connection lost! Reconnect to available node.");
+      ret = _mqtt_hybrid_direct_connection (eh);
```

Comment:
Does _mqtt_hybrid_direct_connection() connect immediately without waiting(lock? ) to receive messages?

Reply:
The lock was added to the connection inside the `_mqtt_hybrid_direct_connection`.
And unlock where waiting for the mqtt message.

Reply:
Suggestion: 
1. lock-unlock when calling _mqtt_hybrid_direct_connection (eh), not in _mqtt_hybrid_direct_connection().
   This function (_nns_edge_message_handler) is thread callback. 
2. call connection-lost event if failed to reconnect.

```
_nns_edge_remove_connection (eh, client_id);
ret = NNS_EDGE_ERROR_CONNECTION_FAILURE;

if (NNS_EDGE_CONNECT_TYPE_HYBRID == eh->connect_type) {
  nns_edge_logi ("Connection lost! Reconnect to available node.");

  nns_edge_lock (eh);
  ret = _mqtt_hybrid_direct_connection (eh);
  nns_edge_unlock (eh);
}

if (ret != NNS_EDGE_ERROR_NONE) {
  nns_edge_event_invoke_callback (eh->event_cb, eh->user_data,
      NNS_EDGE_EVENT_CONNECTION_CLOSED, NULL, 0, NULL);
}
```

Reply:
I reverted lock because this is called inside the thread and call connection-lost event if failed to reconnect. Thanks.